code/game/objects/items/rcd/RPD.dm

```diff
@@ -637,19 +661,24 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 ///Changes the piping layer when the mousewheel is scrolled up or down.
 /obj/item/pipe_dispenser/proc/mouse_wheeled(mob/source_mob, atom/A, delta_x, delta_y, params)
 	SIGNAL_HANDLER
+	single_layer = !(pipe_layers & (pipe_layers - 1))
+	if(!single_layer)
+		to_chat(source_mob, span_notice("Set to single layer first."))
+		return
 	if(source_mob.incapacitated(IGNORE_RESTRAINTS|IGNORE_STASIS))
 		return
 	if(source_mob.get_active_held_item() != src)
 		return
 
+	var/pipe_layer_num = pipe_layer_bits2num(pipe_layers)
 	if(delta_y < 0)
-		piping_layer = min(PIPING_LAYER_MAX, piping_layer + 1)
+		pipe_layers = PIPE_LAYER(min(5, pipe_layer_num + 1))
 	else if(delta_y > 0)
-		piping_layer = max(PIPING_LAYER_MIN, piping_layer - 1)
+		pipe_layers = PIPE_LAYER(max(1, pipe_layer_num - 1))
 	else //mice with side-scrolling wheels are apparently a thing and fuck this up
 		return
 	SStgui.update_uis(src)
-	to_chat(source_mob, span_notice("You set the layer to [piping_layer]."))
+	to_chat(source_mob, span_notice("You set the layer to [pipe_layer_bits2num(pipe_layers)]."))
```

Comment:
Should change to a balloon alert (this is one of the reasons the scroll behavior is crap right now)